backend/src/routes/getBalance.ts

```diff
@@ -0,0 +1,34 @@
+import { FastifyInstance } from 'fastify';
+import { eq } from 'drizzle-orm/pg-core/expressions';
+
+import { usdcBalance } from '@/db/schema';
+
+const addressRegex = /^0x0[0-9a-fA-F]{63}$/;
+
+export function getBalanceRoute(fastify: FastifyInstance) {
+  fastify.get('/get_balance', async (request, reply) => {
+    const { address } = request.query as { address?: string };
+    if (!address) {
+      return reply.status(400).send({ error: 'Address is required.' });
+    }
+    // Validate address format
+    if (!addressRegex.test(address)) {
+      return reply.status(400).send({ error: 'Invalid address format.' });
+    }
```

Comment:
I think this is good for now, once we have more endpoints it may be worth looking into [the zod type provider](https://github.com/turkerdev/fastify-type-provider-zod).